2023-09-29 07:44:57 | INFO    | __main__:from_epic:40 | run - {'stage': 'Initialization EpicPlayer'}
2023-09-29 07:45:01 | DEBUG    | hcaptcha_challenger.onnx.modelhub:request_resource:46 | Downloading resource - {'url': 'https://raw.githubusercontent.com/QIN2DIM/hcaptcha-challenger/main/src/objects.yaml', 'to': '/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/hcaptcha_challenger/onnx/models/objects.yaml'}
2023-09-29 07:45:01 | INFO    | hcaptcha_challenger.onnx.modelhub:flush_runtime_assets:137 | Reloading the local cache of Assets - {'assets_dir': '/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/hcaptcha_challenger/onnx/models/_assets'}
2023-09-29 07:45:01 | DEBUG    | hcaptcha_challenger.onnx.modelhub:flush_runtime_assets:151 | Pulling Assets index file - {'url': 'https://api.github.com/repos/QIN2DIM/hcaptcha-challenger/releases'}
2023-09-29 07:45:01 | INFO    | __main__:claim_epic_games:86 | Try to flush cookie - {'task': 'claim_epic_games'}
2023-09-29 07:45:06 | INFO    | services.agents.epic_games:_login:94 | login - {'url': 'https://www.epicgames.com/id/login?lang=en-US&noHostRedirect=true&redirectUrl=https://store.epicgames.com/en-US/free-games'}
2023-09-29 07:45:06 | INFO    | services.agents.epic_games:_login:96 | login-with-epic - {'url': 'https://www.epicgames.com/id/login/epic?lang=en-US&redirect_uri=https%3A%2F%2Fstore.epicgames.com%2Fen-US%2Ffree-games'}
2023-09-29 07:45:11 | DEBUG    | services.agents.epic_games:_login:108 | claim_weekly_games - {'action': 'handle challenge'}
2023-09-29 07:45:15 | DEBUG    | services.agents.epic_games:_login:111 | handle challenge - {'result': 'retry'}
2023-09-29 07:45:26 | DEBUG    | services.agents.epic_games:_login:111 | handle challenge - {'result': 'retry'}
2023-09-29 07:45:37 | DEBUG    | services.agents.epic_games:_login:111 | handle challenge - {'result': 'retry'}
2023-09-29 07:45:49 | DEBUG    | services.agents.epic_games:_login:111 | handle challenge - {'result': 'retry'}
2023-09-29 07:46:00 | DEBUG    | services.agents.epic_games:_login:111 | handle challenge - {'result': 'retry'}
2023-09-29 07:46:08 | DEBUG    | services.agents.epic_games:_login:111 | handle challenge - {'result': 'retry'}
2023-09-29 07:46:19 | DEBUG    | services.agents.epic_games:_login:111 | handle challenge - {'result': 'retry'}
2023-09-29 07:46:42 | DEBUG    | services.agents.epic_games:_login:111 | handle challenge - {'result': 'retry'}
2023-09-29 07:46:54 | DEBUG    | services.agents.epic_games:_login:111 | handle challenge - {'result': 'retry'}
2023-09-29 07:47:02 | DEBUG    | services.agents.epic_games:_login:111 | handle challenge - {'result': 'retry'}
2023-09-29 07:47:11 | DEBUG    | services.agents.epic_games:_login:111 | handle challenge - {'result': 'retry'}
2023-09-29 07:47:32 | DEBUG    | services.agents.epic_games:_login:111 | handle challenge - {'result': 'retry'}
2023-09-29 07:47:39 | DEBUG    | services.agents.epic_games:_login:111 | handle challenge - {'result': 'retry'}
2023-09-29 07:47:48 | DEBUG    | services.agents.epic_games:_login:111 | handle challenge - {'result': 'retry'}
2023-09-29 07:47:57 | DEBUG    | services.agents.epic_games:_login:111 | handle challenge - {'result': 'retry'}
2023-09-29 07:47:57 | SUCCESS    | services.agents.epic_games:_login:129 | login - {'result': 'token has not expired'}
2023-09-29 07:48:07 | SUCCESS    | services.agents.epic_games:flush_token:150 | flush_token - {'path': PosixPath('/home/runner/work/epic_weekly/epic_weekly/epic/user_data_dir/epic-games@jackylai260707/ctx_cookie.json')}
2023-09-29 07:48:07 | WARNING    | services.agents.epic_games:get_order_history:301 | Failed to get order history, cookie may have expired - {}
2023-09-29 07:48:07 | DEBUG    | __main__:create_tasks:55 | Put task - {'title': 'Model Builder', 'url': 'https://store.epicgames.com/en-US/p/model-builder-f9b456'}
2023-09-29 07:48:07 | DEBUG    | __main__:create_tasks:55 | Put task - {'title': 'Soulstice', 'url': 'https://store.epicgames.com/en-US/p/soulstice'}
2023-09-29 07:48:07 | INFO    | services.agents.epic_games:claim_weekly_games:164 | claim_weekly_games - {'action': 'go to store', 'url': 'https://store.epicgames.com/en-US/p/model-builder-f9b456'}
2023-09-29 07:48:18 | ERROR    | __main__:run:126 | An error has been caught in function 'run', process 'MainProcess' (3513), thread 'MainThread' (140628734159744): - {}
Traceback (most recent call last):
  File "/home/runner/work/epic_weekly/epic_weekly/epic/src/claim.py", line 130, in <module>
    asyncio.run(run())
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py", line 636, in run_until_complete
    self.run_forever()
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py", line 603, in run_forever
    self._run_once()
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py", line 1909, in _run_once
    handle._run()
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
> File "/home/runner/work/epic_weekly/epic_weekly/epic/src/claim.py", line 126, in run
    await agent.stash()
  File "/home/runner/work/epic_weekly/epic_weekly/epic/src/claim.py", line 120, in stash
    await self.claim_epic_games(context)
  File "/home/runner/work/epic_weekly/epic_weekly/epic/src/claim.py", line 102, in claim_epic_games
    await epic.claim_weekly_games(page, self.promotions)
  File "/home/runner/work/epic_weekly/epic_weekly/epic/src/services/agents/epic_games.py", line 180, in claim_weekly_games
    await expect(cta_btn).to_have_text("View In Cart")
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/async_api/_generated.py", line 20041, in to_have_text
    await self._impl_obj.to_have_text(
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/_impl/_assertions.py", line 455, in to_have_text
    await self._expect_impl(
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/playwright/_impl/_assertions.py", line 70, in _expect_impl
    raise AssertionError(f"{out_message}\nActual value: {actual} {log}")
AssertionError: Locator expected to have text 'View In Cart'
Actual value: None 
Call log:
LocatorAssertions.to_have_text with timeout 5000ms
waiting for locator("xpath=//aside//button[@data-testid='add-to-cart-cta-button']")
